flow/directives.py

```diff
@@ -331,7 +335,7 @@ def _is_fraction(value):
 _natural_number = _OnlyType(int, postprocess=_raise_below(1))
 _nonnegative_int = _OnlyType(int, postprocess=_raise_below(0))
 _nonnegative_real = _OnlyType(float, postprocess=_raise_below(0))
-_positive_real = _OnlyType(float, postprocess=_raise_below(1e-12))
+_positive_real = _OnlyType(float, type(None), postprocess=_raise_below(1e-12, True))
```

Comment:
I would add a comment that 1e-12 is an arbitrarily chosen minimum threshold for what constitutes 0.